src/V3LinkJump.cpp

```diff
@@ -228,6 +229,8 @@ class LinkJumpVisitor final : public VNVisitor {
     void visit(AstWhile* nodep) override {
         // Don't need to track AstRepeat/AstFor as they have already been converted
         if (!m_unrollFull.isDefault()) nodep->unrollFull(m_unrollFull);
+        if (m_modp->hasParameterList() || m_modp->hasGParam())
+            nodep->fileline()->modifyWarnOff(V3ErrorCode::UNUSEDLOOP, true);
```

Comment:
This seems overly lax, but given my restrictions I don't have a better suggestion.